src/components/Navigation/AuthenticatedNavigationView.js

```diff
@@ -43,6 +64,112 @@ export default function AuthenticatedNavigationView({ UserNavigationOptions }) {
         setMobileOpen((open) => !open);
     }, []);
 
+    function refreshTokenSetup(res) {
+        console.log('Refresh Token runs');
+        let refreshTiming = (res.tokenObj.expires_in || 3600 - 5 * 60) * 1000;
+
+        const refreshToken = async () => {
+            const newAuthRes = await res.reloadAuthResponse();
+            console.log('newAuthRes:', newAuthRes);
+            console.log('new auth Token ', newAuthRes.id.token);
+
+            setTimeout(refreshToken, refreshTiming);
+        };
+
+        setTimeout(refreshToken, refreshTiming);
+    }
+
+    useEffect(() => {
+        getCourses();
+    }, []);
+
+    useEffect(() => {
+        if (
+            email !== null &&
+            sessionStorage.getItem('google_access_token') === null
+        ) {
+            setGoogleLoginPromptOpen(true);
+        } else {
+            console.log(
+                'Use Effect: ',
+                sessionStorage.getItem('google_access_token')
+            );
+        }
+    }, [email]);
+
+    function getCourses() {
```

Comment:
put this function higher up before it's called in the `useEffect()` on line 82 to help the reader